src/content/blog/2024-12-04-differences-between-linting-and-type-checking.md

```diff
@@ -0,0 +1,274 @@
+// img elements must have an alt prop, either with meaningful text, or an empty string for decorative images.
+```
+
+By adding in rules from plugins, ESLint configurations can be tailored to the specific best practices and common issues to the frameworks a project is built with.
+
+### Areas of Overlap
+
+ESLint and TypeScript operate differently and specialize in different areas of code defects.
+Some code defects straddle the line between "best practices" and "type safety", and so can be caught by both linting and type checking.
+Those defects can often be caught by both ESLint and TypeScript.
+
+To make the most of both tools, we recommend:
+
+* In your ESLint configuration file, use the [ESLint `js.configs.recommended` config](https://eslint.org/docs/latest/use/getting-started#configuration), at least the [`tseslint.configs.recommended` config from typescript-eslint](https://typescript-eslint.io/getting-started/typed-linting), and any community plugins relevant to your project's libraries and frameworks
+* In your TypeScript configuration file, enable [`strict` mode](https://www.typescriptlang.org/tsconfig/#strict) to catch as many type safety issues as possible
+
+typescript-eslint's recommended preset configs disable core ESLint rules that are not helpful with TypeScript.
+The configs leave on any core ESLint rules that are useful alongside type checking.
+
+#### Unused Locals and Parameters
+
+The only TypeScript compiler options we recommend keeping off when using linting are those that enable checking for unused variables:
+
+* [`noUnusedLocals`](https://www.typescriptlang.org/tsconfig/#noUnusedLocals): which reports on unused declared local variables
+* [`noUnusedParameters`](https://www.typescriptlang.org/tsconfig/#noUnusedParameters): which reports on unused function parameters
+
+Those compiler options are useful when not using a linter.
+However, they aren't configurable the way lint rules are, and so can't be configured to higher or lower strictness levels per a project's preferences.
+The compiler options are hardcoded to always ignore any variable whose name begins with `_`.
+
+As an example, the following `registerCallback` function declares two parameters for its callbacks -`id` and `message`- but the developer using it only needed `message`.
+TypeScript's `noUnusedParameters` compiler option would not flag the unused parameter `_`:
+
+```ts
+type Callback = (id: string, message: string) => void;
+
+declare function registerCallback(callback: Callback): void;
+
+// We only want to log message, not id
+registerCallback((_, message) => console.log(message));
+```
+
+Unused variables in JavaScript can also be caught by ESLint's [`no-unused-vars`](https://eslint.org/docs/latest/rules/no-unused-vars) rule; when in TypeScript code, the [`@typescript-eslint/no-unused-vars`](https://typescript-eslint.io/rules/no-unused-vars) is preferable instead.
+The lint rules by default also ignore variables whose name begins with `_`.
```

Comment:
Which lint rules are you referring to? `no-unused-vars` does not ignore variables that begin with `_` by default. You need to use `varsIgnorePattern` and associated options to enable that: https://eslint.org/docs/latest/rules/no-unused-vars#argsignorepattern